src/core/server/uuid/resolve_uuid.test.ts

```diff
@@ -160,39 +198,111 @@ describe('resolveInstanceUuid', () => {
     });
   });
 
+  describe('when file is present with 7.6.0 UUID', () => {
```

Comment:
I think this is the right thing to do to fix the monitoring data for HA Kibana deployments, but to be clear this means all users who install 7.6.0 and then upgrade will get new instance UUIDs for the same instances after upgrading. Seems like an acceptable tradeoff to me?